pkg/controllers/user/logging/utils/embedded.go

```diff
@@ -703,6 +697,46 @@ func newKibanaDeployment(namespace string) *v1beta2.Deployment {
 	return deployment
 }
 
+func UpdateEmbeddedEndpointWithRetry(deploymentLister rv1beta2.DeploymentLister, podLister rv1.PodLister, serviceLister rv1.ServiceLister, clusterLoggings v3.ClusterLoggingInterface, nodeLister v3.NodeLister, clusterName string, logger logstream.LoggerStream) error {
+	timeout := time.After(3 * time.Minute)
+	ticker := time.NewTicker(10 * time.Second)
+
+	wg := sync.WaitGroup{}
+	wg.Add(1)
+	updated := false
+	errCh := make(chan error, 18)
+	go func() {
+		defer wg.Done()
+		for {
+			select {
+			case <-ticker.C:
```

Comment:
we should pass ctx to ticker somehow. So when rancher goes down, ticker has to be notified. You can find some examples in Rancher code on how to do it.